miio/device.py

```diff
@@ -242,9 +243,13 @@ def buttons(self) -> List[ButtonDescriptor]:
         """Return a list of button-like, clickable actions of the device."""
         return []
 
-    def settings(self) -> Dict[str, SettingDescriptor]:
+    def settings(
+        self,
+    ) -> Dict[str, Union[EnumSettingDescriptor, NumberSettingDescriptor]]:
         """Return list of settings."""
-        settings = self.status().settings()
+        settings = (
+            self.status().settings()
+        )  # NOTE that this already does IO so schould be run in executer job in HA
```

Comment:
I can move all the binding/IO (status retrieving, choices list retrieving etc) to the `info()` call, cache it there and then return the list of sensors/buttons/settings from the cache.
If sensors/buttons/settings is accesed when cache is still empty I can call `info()` there.

Do you want me to move it?

Reply:
Or we could raise an error when sensors/buttons/settings is accesed when cache is still empty, maybe that is nicer since you then know that the sensors/buttons/settings will not do IO, and therefore do not have to be run in a executor job...